vagrant/ansible/roles/client.prep/tasks/main.yml

```diff
@@ -2,6 +2,11 @@
   include_tasks:
     file: "{{ distro.actions_file }}"
 
+- name: Process infrastructure specific tasks for clients
+  include_role:
+    name: 'sit.{{ filesystem }}'
```

Comment:
I checked the rest of the commit and couldn't find where the filesystem was define but from your earlier comments for this issue, I realised that we have to pass a new EXTRA_VARS for this to work properly. 
This was later renamed backend and the EXTRA_VARS is automatically set in the makefile. 

There are a few issues I have with this.
1) There will be a few commits where we cannot simply do a make test and it just works. This isn't much of an issue with this project and maybe we can ignore for this case. But I like to see continuity in patches.
2) Why not set backend in vagrant/roles/local.defaults/tasks instead of the Makefile? This would be consistent with how we set the distro variable.

Reply:
> I checked the rest of the commit and couldn't find where the filesystem was define but from your earlier comments for this issue, I realised that we have to pass a new EXTRA_VARS for this to work properly. This was later renamed backend and the EXTRA_VARS is automatically set in the makefile.
> 
> There are a few issues I have with this.
> 
>     1. There will be a few commits where we cannot simply do a make test and it just works. This isn't much of an issue with this project and maybe we can ignore for this case. But I like to see continuity in patches.

I can squash all the commits into a single one if you prefer. I decided to make the changes once reviews had started so I thought that it would be easier for reviewers to apply the changes as new commits, but I'm ok to convert it to a single commit.

> 
>     2. Why not set backend in vagrant/roles/local.defaults/tasks instead of the Makefile? This would be consistent with how we set the distro variable.

As I see it, the idea is to make it very flexible by easily switching the OS or backend. Defining values inside a yaml of a role doesn't seem very flexible to me. In fact, in another PR I'm preparing I've moved the distro selection outside the role and it could be selected by environment variables or directly from `make` command like this:

    # make gf c7 (Build the test environment on CentOS 7 using GlusterFS)
or

    # make ceph c8 (Build the test environment on CentOS 8 using CephFS)

Once built, the configuration is stored and a simple `make test` should work without explicitly specifying the environment.

Is that ok ?

Reply:
1) I prefer to keep it as separate commits. So I guess we can leave it as it is for now.

2) By setting the field in vagrant/roles/local.defaults/tasks, you just set the default values to use if the value isn't defined in the EXTRA_VARS. 
example: To test a centos 7 build, we use
EXTRA_VARS="use_distro=centos7" make
This overrides the use_distro mentioned in the local.defaults.
see: https://github.com/samba-in-kubernetes/sit-environment/blob/main/docs/Fedora-Setup.md

Reply:
From outside (being the first time one looks into this repo) it seems easier to find where default values are defined if they are near the command the user executes (i.e. inside the `Makefile` that `make` loads). Identifying that the defaults are set in a conditional task (not even a variables file) of a specific role two levels below the main directory is not obvious. That's why I didn't use the same approach for `backend`.

In any case, I'll modify it to use the same approach now.